core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/part/ConstrainedEnvelopePartBuilder.java

```diff
@@ -102,7 +104,7 @@ public boolean addStep(double position, double speed, double timeDelta) {
             position = intersection.position;
             speed = intersection.speed;
             if (position == lastPos) {
```

Comment:
Wouldn't `arePositionsEqual` make sense here ?

Reply:
The envelopes mostly work with actual equalities when it comes to positions, if we add a tolerance here we would need other changes